modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java

```diff
@@ -286,6 +304,156 @@ public Response getAsset(@PathParam("mediaPackageID") final String mediaPackageI
+          })
+  public Response getWorkflowProperties(@PathParam("mediaPackageID") final String mediaPackageId) {
+    try {
+      final AQueryBuilder queryBuilder = getAssetManager().createQuery();
+      final AResult result = queryBuilder.select(queryBuilder.propertiesOf(WORKFLOW_PROPERTIES_NAMESPACE))
+              .where(queryBuilder.mediaPackageId(mediaPackageId).and(queryBuilder.version().isLatest())).run();
+
+      // we expect exactly one result when specifying a media package id
+      if (result.getSize() < 1) {
+        return notFound();
+      } else if (result.getSize() > 1) {
+        return serverError();
+      }
+
+      // build map from properties
+      HashMap<String, String> properties = new HashMap<>();
+      if (result.getRecords().head().isSome()) {
+        for (final Property property : result.getRecords().head().get().getProperties()) {
+          properties.put(property.getId().getName(), property.getValue().get(Value.STRING));
+        }
+      }
+      return ok(gson.toJson(properties));
+    } catch (Exception e) {
+      return handleException(e);
+    }
+  }
+
+
+  @POST
+  @Path("{mediaPackageID}/workflowProperties")
+  @RestQuery(name = "detWorkflowProperties",
+          description = "Set additional workflow properties",
+          pathParameters = {
+                  @RestParameter(
+                          name = "mediaPackageID",
+                          description = "the media package ID",
+                          isRequired = true,
+                          type = STRING)
+          },
+          restParameters = {
+                  @RestParameter(
+                          name = "properties",
+                          isRequired = true,
+                          type = STRING,
+                          description = "JSON object containing new properties")
+          },
+          reponses = {
+                  @RestResponse(description = "Properties successfully set", responseCode = SC_CREATED),
+                  @RestResponse(description = "Invalid data", responseCode = SC_BAD_REQUEST),
+                  @RestResponse(description = "Internal error", responseCode = SC_INTERNAL_SERVER_ERROR) },
+          returnDescription = "Returned status code indicates success")
+  public Response setWorkflowProperties(@PathParam("mediaPackageID") final String mediaPackageId,
+          @FormParam("properties") final String propertiesJSON) {
+    Map<String, String> properties;
+    try {
+      properties = gson.fromJson(propertiesJSON, stringMapType);
+    } catch (Exception e) {
+      return badRequest();
+    }
+    for (final Map.Entry<String, String> entry : properties.entrySet()) {
+      final PropertyId propertyId = PropertyId.mk(mediaPackageId, WORKFLOW_PROPERTIES_NAMESPACE, entry.getKey());
+      final Property property = Property.mk(propertyId, Value.mk(entry.getValue()));
+      if (!getAssetManager().setProperty(property)) {
+        return notFound();
```

Comment:
Why "not found"? Shouldn't the property be created here?

Reply:
No. `setProperty()` will return `false` if the event identified by the media package id does not exist in the asset manager. Well… that's the idea (and documentation) at least. There are a bunch of issues in the asset manager implementation related to this… but at least it's very likely the case ;-P